endpoints/openrtb2/amp_auction.go

```diff
@@ -578,23 +580,24 @@ func setAmpExt(site *openrtb2.Site, value string) {
 	}
 }
 
-func readPolicy(consent string) (privacy.PolicyWriter, error) {
-	if len(consent) == 0 {
-		return privacy.NilPolicyWriter{}, nil
-	}
-
-	if gdpr.ValidateConsent(consent) {
-		return gdpr.ConsentWriter{consent}, nil
+// setRegExtGDPR sets regs.ext.gdpr to either 0 or 1 only if it was set in the request query
+func setRegExtGDPR(gdprApplies *bool, req *openrtb2.BidRequest) error {
+	if gdprApplies == nil {
+		return nil
 	}
 
-	if ccpa.ValidateConsent(consent) {
-		return ccpa.ConsentWriter{consent}, nil
+	gdpr := int8(0)
+	if *gdprApplies {
+		gdpr++
```

Comment:
IMHO: It's better to set this to explicitly 1 instead of using the increment helper, since the value 1 has a specific  meaning.

Reply:
Done